pkg/utils/openshift/utils.go

```diff
@@ -48,6 +51,69 @@ func MapKnownVersion(info platform.PlatformInfo) platform.OpenShiftVersion {
 		"1.10+": "3.10",
 		"1.11+": "3.11",
 		"1.13+": "4.1",
+		"1.14+": "4.2",
+		"1.16+": "4.3",
 	}
 	return platform.OpenShiftVersion{Version: k8sToOcpMap[info.K8SVersion]}
 }
+
+/*
+Supported OpenShift platform
+Supported version format : Major.Minor.Patch
+	e.g.: 2.3.4
+ */
+func CompareOpenShiftVersion(cfg *rest.Config, version string) (int, error) {
+	isOcp, err := IsOpenShift(cfg)
+	if err != nil {
+		return -1, err
+	}
+	if !isOcp {
+		return -1, errors.New("There is no OpenShift platform detected.")
+	}
+	info, err := GetPlatformInfo(cfg)
+	if err != nil {
+		return -1, err
+	}
+	curVersion := MapKnownVersion(info)
+	return CompareVersion(curVersion.Version, version)
+}
+
+/*
+Supported version format : Major.Minor.Patch
+e.g.: 2.3.4
+return:
+	-1 : if ver1 < ver2
+	 0 : if ver1 == ver2
+     1 : if ver1 > ver2
+The int value returned should be discarded if err is not nil
+*/
+func CompareVersion(ver1 string, ver2 string) (int, error) {
+	ver1Nums := strings.Split(ver1, ".")
+	ver2Nums := strings.Split(ver2, ".")
+	length := len(ver1Nums)
+	if length < len(ver2Nums) {
+		length = len(ver2Nums)
+	}
+	for i := 0; i < length; i++ {
+		v1 := 0
+		if i < len(ver1Nums) {
+			if v1, err := strconv.Atoi(ver1Nums[i]); err != nil {
+				_ = v1
+				return -1, err
+			}
+		}
+		v2 := 0
+		if i < len(ver2Nums) {
+			if v2, err := strconv.Atoi(ver2Nums[i]); err != nil {
+				_ = v2
+				return -1, err
+			}
+		}
+		if v1 > v2 {
+			return 1, nil
+		} else if v2 > v1 {
+			return -1, nil
+		}
+	}
+	return 0, nil
+}
```

Comment:
We can use the OpenShiftVersion struct instead of strings for params here and make the func more readable and self-documenting (note that I have not tested this, you'll need to do so and also add test cases):
```
func CompareOpenShiftVersion(ver1 OpenShiftVersion, ver2 OpenShiftVersion) (int, error) {

	if d, err := compareSegment(ver1.MajorVersion(), ver2.MajorVersion()); d != 0 {
		return d, err
	}
	if d, err := compareSegment(ver1.MinorVersion(), ver2.MinorVersion()); d != 0 {
		return d, err
	}
	if d, err := compareSegment(ver1.BuildVersion(), ver2.BuildVersion()); d != 0 {
		return d, err
	}
	// major/minor/build all match
	return 0, nil
}

func compareSegment(str1 string, str2 string) (int, error) {
	var ver1, ver2 int
	var err error

	if ver1, err = strconv.Atoi(str1); err != nil {
		return -1, err
	}
	if ver2, err = strconv.Atoi(str2); err != nil {
		return -1, err
	}
	if ver1 < ver2 {
		return -1, nil
	}
	if ver1 > ver2 {
		return 1, nil
	}
	return 0, nil
}
```

Reply:
I didn't change anything about your CompareVersion func parameters, just the func name... it still accepts a string as it did before. If you apply all of the comments and follow the code, you'll see that.

Reply:
I think I see the issue... you're confusing two different funcs b/c of the name change I proposed b/c now the secondary func matches the name of your initial func...

Was:
CompareOpenShiftVersion calls --> compareVersion

What I Recommend:
CompareOpenShiftVersion calls --> compareVersion calls --> compareSegment. 

Code in comment updated to reflect.

Reply:
modified.
- get the know that openshfit pkg is acting like a proxy to platform!
I moved the new method to platform pkg to make things right, and this also makes creating test cases possible.
- the OpenShfitVersion is not meant to handle semantic version.   its buildversion is string not the same as semver which is defined as numeric.  With the existing impl., only Major and Minor are compared.